content/data-loading.md

```diff
@@ -0,0 +1,556 @@
+
+## Defining a publication
+
+A publication should be defined in a server-only file. For instance, in the Todos example app, we want to publish the set of public lists to all users:
+
+```js
+Meteor.publish('lists/public', function() {
+  return Lists.find({userId: {$exists: false}});
+});
+```
+
+There are a few things to understand about this code block. Firstly, we've named the publication `lists/public`, and that will be how we access it from the client. Secondly, we are simply returning a Mongo *cursor* from the publication function.
+
+What that means is that the publication will simply ensure the set of data matching that query is available to any client that subscribes to it. In this case, all lists that do not have a `userId` setting. So the collection named `'Lists'` on the client (which we called `Lists` of course) will have all of the public lists that are available in the server collection named `'Lists'` whilst that subscription is open (which is all the time in our example app).
+
+There are two types of parameters of a publish function. Firstly, a publication always has the current user's `id` available at `this.userId` (this is why we use the `function() {}` form for publications rather than the ecmascript `() => {}`. You can disable the linting rule for publication files with `eslint-disable prefer-arrow-callback`).
+
+```js
+Meteor.publish('lists/private', function() {
+  if (!this.userId) {
+    return this.ready();
+  }
+
+  return Lists.find({userId: this.userId});
+});
+```
+
+What this means is that the above publication can be confident (thanks to Meteor's accounts system) that it will only ever publish private lists to the user that they belong to. Note also that the publication will re-run if the user logs out (or back in again), which means that the published set of private lists will reflect this.
+
+In the case of a non-logged in user, we explicitly call `this.ready()`, which indicates to the subscription that we've sent all the data we are initially going to send (in this case none).
+
+The other type of publication argument is a simple named argument:
+
+```js
+Meteor.publish('list/todos', function(listId) {
+  check(listId, String);
+
+  ...
+});
+```
+
+When we create a subscription to this piblication on the client, we can provide this argument via the `Meteor.subscribe()` call:
+
+```js
+Meteor.subscribe('list/todos', list._id);
+```
+
+Also note that we are careful to check the `listId` is a string as we expect, as described in the Security article.
```

Comment:
This should be co-located with the code. Perhaps this could even be a comment. We'll have a whole bit on this in the security article.